lib/interceptor.js

```diff
@@ -422,6 +422,11 @@ Interceptor.prototype.basicAuth = function basicAuth(options) {
 Interceptor.prototype.query = function query(queries) {
     this.queries = this.queries || {};
 
+    if (queries && _.isFunction(queries) && queries() === false) {
+      this.queries = {}
+      return this
+    }
```

Comment:
Can you make a separate PR for the code change and only do tests? It's harder to review that way

Reply:
Sure!

Also, care to guess the expected behavior when function returns false in the query function? Just so I don't misinterpret the results I'm getting.

Reply:
I don’t know 🤷‍♀️ Can you open a separate issue/PR for discussion?

Reply:
Yeah. Once this test is fixed I'll work on it. :)

Reply:
@jlilja Any time to work on this? :)